### PR TITLE
[Merged by Bors] - chore(Data/Funlike): update examples and replace Lean 3 syntax

### PR DESCRIPTION
**Pull request by grunweg**
Fully update the module docstrings (in particular, the examples given therein) after #8386.

This includes switching to where syntax, but also replacing Lean 3 syntax, replacing => by "\mapsto" while at it and indenting code per the style guide. As such, it's also a follow-up to #11301.

Co-authored-by: @Vierkantor

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/FunLike/Equiv.lean [hunk 1]
```diff
@@ -84,27 +83,26 @@ The second step is to add instances of your new `MyIsoClass` for all types exten
...
 
 ```
-structure CoolerIso (A B : Type*) [CoolClass A] [CoolClass B]
-  extends MyIso A B :=
-(map_cool' : toFun CoolClass.cool = CoolClass.cool)
+structure CoolerIso (A B : Type*) [CoolClass A] [CoolClass B] extends MyIso A B :=
+  (map_cool' : toFun CoolClass.cool = CoolClass.cool)
 
 section
 set_option old_structure_cmd true
```
Comment (grunweg) [inline Mathlib/Data/FunLike/Equiv.lean:90]: This option doesn't any more either: can this be removed or should it be updated?

File: Mathlib/Data/FunLike/Embedding.lean [hunk 2]
```diff
@@ -86,29 +87,29 @@ The second step is to add instances of your new `MyEmbeddingClass` for all types
...
 
 ```
-structure CoolerEmbedding (A B : Type*) [CoolClass A] [CoolClass B]
-  extends MyEmbedding A B :=
-(map_cool' : toFun CoolClass.cool = CoolClass.cool)
+structure CoolerEmbedding (A B : Type*) [CoolClass A] [CoolClass B] extends MyEmbedding A B :=
+  (map_cool' : toFun CoolClass.cool = CoolClass.cool)
 
 section
 set_option old_structure_cmd true
```
Comment (grunweg) [inline Mathlib/Data/FunLike/Embedding.lean:94]: This option doesn't exist in Lean 4: can this be removed or should it be updated?

**Comment by dupuisf** [comment]
Here, there is an additional issue: the whole `FunLike` system was recently refactored, and it seems like these docstrings did not follow suit.  @Vierkantor, could you have a look at this to make sure this documentation is updated correctly?

**Comment by Vierkantor** [comment]
Thank you for letting me know, I totally forgot about updating the `EmbeddingLike` docs. I think it's best if I push the fixes to this branch directly, since they are so extensive. Feel free to modify and revert if needed.

**Comment by grunweg** [comment]
Thank you for the thorough update! I have just merged master and updated the PR description.

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors r+

**Pull request closed without merging**